core/backend/src/SQLiteDb.ts

```diff
@@ -6,58 +6,141 @@
+import { CloudSqlite, IModelJsNative } from "@bentley/imodeljs-native";
 import { DbResult, IDisposable, OpenMode } from "@itwin/core-bentley";
+import { LocalFileName } from "@itwin/core-common";
 import { IModelHost } from "./IModelHost";
 import { SqliteStatement, StatementCache } from "./SqliteStatement";
 
+/** @public */
+export namespace SQLiteDb {
+  /** A CloudSqlite container that may be connected to a CloudCache.
+   * @internal
+   */
+  export type CloudContainer = IModelJsNative.CloudContainer;
+  /** @internal */
+  export type CloudCache = IModelJsNative.CloudCache;
+  /** @internal */
+  export type CloudPrefetch = IModelJsNative.CloudPrefetch;
+
+  /** Parameters for creating a new SQLiteDb
+   * @internal
+   */
+  export type CreateParams = IModelJsNative.SQLiteDbCreateParams;
+
+  /** Parameters for opening a SQLiteDb
+   * @internal
+   */
+  export type OpenParams = IModelJsNative.SQLiteDbOpenParams;
+
+  /** Incremental IO for blobs
+   * @internal
+   */
+  export type BlobIO = IModelJsNative.BlobIO;
+}
+
 /** A SQLiteDb file
  * @public
  */
 export class SQLiteDb implements IDisposable {
-  private _nativeDb?: IModelJsNative.SQLiteDb;
-  private _sqliteStatementCache = new StatementCache<SqliteStatement>();
-
   /** @internal */
-  public get nativeDb(): IModelJsNative.SQLiteDb {
-    return this._nativeDb!; // eslint-disable-line @typescript-eslint/no-non-null-assertion
-  }
+  public readonly nativeDb = new IModelHost.platform.SQLiteDb();
+  private _sqliteStatementCache = new StatementCache<SqliteStatement>();
 
-  constructor() {
-    this._nativeDb = new IModelHost.platform.SQLiteDb();
-  }
-  /** Call this function when finished with this SQLiteDb. This releases the native resources */
+  /** alias for closeDb. */
   public dispose(): void {
-    if (!this._nativeDb)
-      return;
-
     this.closeDb();
-    this._nativeDb.dispose();
-    this._nativeDb = undefined;
   }
 
   /** Create a SQLiteDb
-   * @param pathName The path to the SQLiteDb file to create.
+   * @param dbName The path to the SQLiteDb file to create.
    */
-  public createDb(pathName: string, container?: IModelJsNative.CloudContainer, params?: IModelJsNative.SQLiteDbCreateParams): void {
-    this.nativeDb.createDb(pathName, container, params);
+  public createDb(dbName: string, container?: SQLiteDb.CloudContainer, params?: SQLiteDb.CreateParams): void {
+    this.nativeDb.createDb(dbName, container, params);
   }
 
   /** Open a SQLiteDb.
-   * @param pathName The path to the SQLiteDb file to open
+   * @param dbName The path to the SQLiteDb file to open
    * @param container optional CloudContainer holding database
    */
-  public openDb(pathName: string, openMode: OpenMode | IModelJsNative.SQLiteDbOpenParams, container?: IModelJsNative.CloudContainer): void {
-    this.nativeDb.openDb(pathName, openMode, container);
+  public openDb(dbName: string, openMode: OpenMode | SQLiteDb.OpenParams, container?: SQLiteDb.CloudContainer): void {
```

Comment:
This `@public` method takes arguments of `@internal` types. Do you intend to move them to core-common? Add an internal method that takes the internal types? Define an `Args` interface type that applies appropriate release tags to the individual fields? Is the SQLiteDb namespace tricking api-extractor into accepting this?
Same for `withOpenDb`.

Reply:
I was surprised that extract-api didn't complain about the internal types, I was sure that it would. It didn't so I went with it.

But this is a public method of a public class, so I can't change the first two arguments. What do you think of this:

```ts
  /** Open a SQLiteDb.
   * @param dbName The path to the SQLiteDb file to open
   */
  public openDb(dbName: string, openMode: OpenMode): void;
  /**
   * @param container optional CloudContainer holding database
   * @internal
   */
  public openDb(dbName: string, openMode: OpenMode | SQLiteDb.OpenParams, container?: SQLiteDb.CloudContainer): void;

  /** @internal */
  public openDb(dbName: string, openMode: OpenMode | SQLiteDb.OpenParams, container?: SQLiteDb.CloudContainer): void {
    this.nativeDb.openDb(dbName, openMode, container);
  }
```

Reply:
And of course this is why it's best to use named arguments from the start.

Reply:
Looks good to me. Hopefully the overload won't trip up the docs generator.